civis/futures.py

```diff
@@ -120,9 +120,15 @@ def __init__(self, poller, poller_args,
                          client=client,
                          poll_on_creation=poll_on_creation)
 
-        if hasattr(client, 'channels'):
-            config, channels = self._pubnub_config()
-            self._pubnub = self._subscribe(config, channels)
+    def _begin_tracking(self):
+        # Be sure to subscribe to the PubNub channel before polling.
+        # Otherwise the job might complete after polling and before
+        # we subscribe, causing us to miss the notification.
+        with self._condition:
+            if hasattr(self.client, 'channels') and not self.subscribed:
+                config, channels = self._pubnub_config()
+                self._pubnub = self._subscribe(config, channels)
+            super()._begin_tracking()
```

Comment:
I think this would be a bit easier to read if there were a comment here like, "The superclass PollableResult manages polling."

Reply:
Comment added.